fah-data-cleanup.pl

```diff
@@ -5,44 +5,55 @@
 use Cwd;
 use Getopt::Long qw(HelpMessage :config pass_through);
 
-my $proj     = $ARGV[0] or die HelpMessage();
-my $maxrun   = $ARGV[1] or die HelpMessage();
-my $maxclone = $ARGV[2] or die HelpMessage();
-
-my $currentrun = 0;
-my $homedir    = getcwd();
-
-while ($currentrun < $maxrun) {
-    my $currentclone = 0;
-    while ($currentclone < $maxclone) {
-        my $workdir = "$homedir/PROJ$proj/RUN$currentrun/CLONE$currentclone/";
-        chdir $workdir;
-        my $test = getcwd();
-        print STDOUT "Working on directory $test ...\n";
-        `rm *# *.xvg *.pdb *.out *.nat *.nat6 temp* 2> /dev/null`;
-        `mv frame0.tpr temp`;
-        `rm *.tpr 2> /dev/null`;
-        `mv temp frame0.tpr`;
-        `mv ener.edr temp`;
-        `rm *.edr 2> /dev/null`;
-        `mv temp ener.edr`;
-        $currentclone++;
+my $projpath = $ARGV[0] or die HelpMessage();
+
+if (-d $projpath) {
+    my @runs = &pattern_walk("RUN", $projpath);
+    foreach my $run (@runs) {
+        my @clones = &pattern_walk("CLONE", $run);
+        foreach my $clone (@clones) {
+            print STDOUT "Working on directory $clone ...\n";
+            chdir $clone;
+            `rm *# *.xvg *.pdb *.out *.nat *.nat6 *.txt temp* 2> /dev/null`;
+            `mv frame0.tpr temp`;
+            `rm *.tpr 2> /dev/null`;
+            `mv temp frame0.tpr`;
+            `mv ener.edr temp`;
+            `rm *.edr 2> /dev/null`;
+            `mv temp ener.edr`;
+        }
     }
-    $currentrun++;
+} else {
+    print STDOUT "$projpath: No such file or directory\n";
+    exit 1;
 }
-
 print STDOUT "Done!\n";
 
+# Arguments: pattern to search for, absolute path of directory
+sub pattern_walk {
+    my ($pattern, $path, @dirs, $num_dirs);
```

Comment:
Use first line in sub body to get parameters: `my ($pattern, $path) = @_;`. This, as a result, tells you your sub's parameters.

Reply:
Please also rename the sub name to better show intention: this sub looks like it tries to **get dirs** that match a pattern. A language's characteristics (e.g. python's `os.walk()`) should not drive how we name things. We need to name things that's easy to understand and clearly shows meaning/intention.